### DISCUSSION TITLE
MIPMAP generation on existing bdv files

### DISCUSSION THREAD

**Discussion by miketaormina** [nvladimus/npy2bdv#8] [Ideas]
Hi Nikita,

I'm wondering if it is possible to create subsampled versions of an already existing volume, or if the file structure constrains this to happen at the time of creation? For example, is it necessary to create a new volume via BdvWriter with the desired image pyramids and use the original to populate it (duplicating data on disk)?

Thanks,
Mike

**Comment by nvladimus** [comment]
Hi, Mike,
Sorry for long delay - I didn't see this message. It is definitely possible to do this, by adding new datasets to existing H5 file, but this feature is not implemented yet. There is a class [BdvEditor](https://nvladimus.github.io/npy2bdv/npy2bdv.html#npy2bdv.npy2bdv.BdvEditor) to do this kind of post-hoc operations, it would be a good place to put this function. Unfortunately I have very little time to implement it in the next few months. but you are welcome to make a PR!

**Comment by nvladimus** [comment]
After some pondering, I implemented this similar feature in **v.1.0.7**, see examples: [5.2 Stacks are written as raw data followed by lazy downsampling and compression.](https://github.com/nvladimus/npy2bdv/blob/master/docs/examples/examples_h5writing.ipynb) and docs:  [.create_pyramids(...)](https://nvladimus.github.io/npy2bdv/npy2bdv.html#npy2bdv.npy2bdv.BdvBase.create_pyramids).

**Reply by miketaormina** [reply]
Awesome, thanks for the effort! This package has been very useful.
